src/main/java/org/teamresistance/frc/Gyro3D.java

```diff
@@ -0,0 +1,46 @@
+package org.teamresistance.frc;
+
+import com.kauailabs.navx.frc.AHRS;
+import edu.wpi.first.wpilibj.SPI;
+import org.strongback.components.AngleSensor;
+
+/**
+ * Created by shrey on 12/18/2016.
+ */
+public class Gyro3D extends AHRS implements AngleSensor {
+
+  public Gyro3D(SPI.Port spi_port_id) {
+    super(spi_port_id);
+  }
+
+
+  @Override
+  public double getAngle() {
+    return getYawVal();
+  }
+
+  public double normalize(double angle) {
+    return ((angle%360)+360)%360;
+  }
+
+  public double getYawVal() {
+    return normalize((double)super.getYaw());
+  }
+
+  public double getRollVal() {
+    return normalize((double)super.getRoll());
+  }
+
+  public double getPitchVal() {
+    return normalize((double)super.getPitch());
+  }
+
+  public boolean isStraight(double deadband, double startAngle) {
```

Comment:
I had originally made this method to be able to hold the angle at some angle x, but considering the functionality can be broadened to check if any angle is on a target angle, i changed the namings for convention purposes.